bindings/python/src/ser.rs

```diff
@@ -74,6 +74,23 @@ fn get_object_type_from_object(object: *mut pyo3::ffi::PyObject) -> ObjectType {
     }
 }
 
+fn get_type_name(object_type: *mut pyo3::ffi::PyTypeObject) -> String {
+    let type_name = unsafe { CStr::from_ptr((*object_type).tp_name).to_string_lossy() };
+    type_name.to_string()
+}
+
+#[inline]
+fn check_type_is_str<E: ser::Error>(object: *mut pyo3::ffi::PyObject) -> Result<(), E> {
+    let object_type = unsafe { Py_TYPE(object) };
+    if object_type != unsafe { types::STR_TYPE } {
+        return Err(ser::Error::custom(format!(
+            "Dict key must be str. Got '{}'",
+            get_type_name(object_type)
```

Comment:
Nit: I think in this case it should be possible to avoid one `String` allocation inside `get_type_name`. See another comment